Modified docs/Documentation/03a-stoplight-flavored-markdown.md (#562)
stoplightio/studio: docs/Documentation/03a-stoplight-flavored-markdown.md

# Stoplight Flavored Markdown (SMD)

## Overview

> Stoplight flavored markdown can be used to enhance existing markdown documentation with necessary documentation components like code blocks, images, diagrams e.t.c. 

Reviewer: ```suggestion
Stoplight Flavored Markdown can be used to enhance existing Markdown documentation with necessary documentation components like code blocks, images, diagrams etc.
```

### Components

- [Callouts](#callouts)
- [Task Lists](#task-lists)
- [Code Blocks](#code-blocks)
- [Code Groups](#code-groups)
- [Images](#images)
- [Graphs and Diagrams](#graphs-and-diagrams)
- [JSON Schema](#json-schema)
- [Tabs](#tabs)
- [HTML Support](#html)
- [Embeds](#embeds)

### The Two Laws

1.  SMD is human readable. A human with a simple text editor can easily read and write SMD.
2.  SMD degrades gracefully. SMD documents rendered on `github.com` should be clean and readable.

### The Approach

> 1.  Stoplight flavored markdown extends GitHub flavor markdown with inline comment annotations.

Reviewer: ```suggestion
1.  Stoplight Flavored Markdown extends GitHub Flavored Markdown with inline comment annotations.
```

> 2.  The value inside of the annotations is a YAML object, and the annotation affects the markdown block that directly follows it in the document.

Reviewer: ```suggestion
2.  The value inside of the annotations is a YAML object, and the annotation affects the Markdown block that directly follows it in the document.
```

> By leveraging comments to store annotations, Stoplight flavored markdown degrades gracefully to any other markdown

Reviewer: ```suggestion
By leveraging comments to store annotations, Stoplight Flavored Markdown degrades gracefully to any other Markdown
```

> renderer (Github, for example).

Reviewer: ```suggestion
renderer (GitHub, for example).
```

Reviewer: GitHub

## Callouts

A callout is an MD blockquote with an optional annotation that indicates intent.

### Danger

<!-- theme: danger -->

> #### Danger Will Robinson!
>
> Here is my danger callout!

```md title="Copy this code to try it out!"
<!-- theme: danger -->

> #### Danger Will Robinson!
>
> Here is my danger callout!
```

### Warning

<!-- theme: warning -->
> #### Watch Out!
>
> Here is my warning callout!

```md title="Copy this code to try it out!"
<!-- theme: warning -->
> #### Watch Out!
>
> Here is my warning callout!
```

### Success

<!-- theme: success -->

> #### Mission Accomplished!
>
> Here is my success callout!

```md title="Copy this code to try it out!"
<!-- theme: success -->

> #### Mission Accomplished!
>
> Here is my success callout!
```

### Info

<!-- theme: info -->

> #### A thing to know
>
> Here is my info callout

```md title="Copy this code to try it out!"
<!-- theme: info -->

> #### A thing to know
>
> Here is my info callout
```

## Task Lists

- [ ] one
- [x] two
- [ ] three

```md title="Copy this code to try it out!"
- [ ] one
- [x] two
- [ ] three
```

## Code Blocks

> A SMD code block is an MD code fence with an optional annotation to tweak the presentation of the code block.

Reviewer: ```suggestion
An SMD code block is an MD code fence with an optional annotation to tweak the presentation of the code block.
```

Reviewer: @mnaumanali94 if you're going to say "an MD", it should also be "An SMD".

```

**Center Focus**

Make screenshots pop out with a center focus and a default background image. 

<!-- focus: center -->

![Dev portal settings](https://i.imgur.com/YCb6MWI.png)

```md title="Try it out!"
<!-- focus: center -->

![Dev portal settings](https://i.imgur.com/YCb6MWI.png)
```

**Add a Caption**

Add an optional caption to explain the screenshot further. 

<!-- focus: top -->

![Studio project share](https://i.imgur.com/ueOOL8X.png 'Can add an optional caption')

```md title="Try it out!"
<!-- focus: top -->

![Studio project share](https://i.imgur.com/ueOOL8X.png 'Can add an optional caption')
```

**Try a Different BG Color**

<!--
focus: top
bg: primary
-->

![Studio project share](https://i.imgur.com/ueOOL8X.png)

```md title="Try it out!"
<!--
focus: top
bg: primary
-->

![Studio project share](https://i.imgur.com/ueOOL8X.png 'Can add an optional caption')
```

**Use a hex BG Color**

<!--
focus: top
bg: "#f78ae0"
-->

![Studio project share](https://i.imgur.com/ueOOL8X.png)

```md title="Try it out!"
<!--
focus: top
bg: "#f78ae0"
-->

![Studio project share](https://i.imgur.com/ueOOL8X.png)
```

## Graphs and Diagrams

> You can in graphs and diagrams in your markdown via the `mermaid` code block language tag. Here are some examples:

Reviewer: You can what?

Reviewer: ```suggestion
You can in graphs and diagrams in your Markdown via the `mermaid` code block language tag. Here are some examples:
```

    end
```

### Sequence

```mermaid
sequenceDiagram
    autonumber
    Alice->>John: Hello John, how are you?
    loop Healthcheck
        John->>John: Fight against hypochondria
    end
    Note right of John: Rational thoughts!
    John-->>Alice: Great!
    John->>Bob: How about you?
    Bob-->>John: Jolly good!
```

```example
sequenceDiagram
    autonumber
    Alice->>John: Hello John, how are you?
    loop Healthcheck
        John->>John: Fight against hypochondria
    end
    Note right of John: Rational thoughts!
    John-->>Alice: Great!
    John->>Bob: How about you?
    Bob-->>John: Jolly good!
```

### Journey

```mermaid
journey
    title My working day
    section Go to work
      Make tea: 5: Me
      Go upstairs: 3: Me
      Do work: 1: Me, Cat
    section Go home
      Go downstairs: 5: Me
      Sit down: 5: Me
```

```example
journey
    title My working day
    section Go to work
      Make tea: 5: Me
      Go upstairs: 3: Me
      Do work: 1: Me, Cat
    section Go home
      Go downstairs: 5: Me
      Sit down: 5: Me
```

## JSON Schema

The SMD JSON schema block is an MD code block with an additional `json_schema` language tag. The contents of the code

> fence should be the JSON schema object to be rendered. The primary language tag can be `YAML, `YML`, or `JSON.

Reviewer: ```suggestion
fence should be the JSON schema object to be rendered. The primary language tag can be `YAML, `YML`, or `JSON`.
```

Reviewer: Did you mean to resolve this? It looks strange if you don't add the extra ` after JSON

  "required": ["id", "name"]
}
```
Try it out with the example below:

```json title="Try it out!" 
```json json_schema
{
  "title": "User",
  "type": "object",
  "properties": {
    "id": {
      "type": "string"
    },
    "name": {
      "type": "string",
      "description": "The user's full name."
    },
    "age": {
      "type": "number",
      "minimum": 0,
      "maximum": 150
    }
  },
  "required": ["id", "name"]
}
```

## Tabs

An SMD tab container is a `tab` annotation, followed by the tab content, and closed by a final `tab-end` annotation.

<!-- theme: danger -->

> Tab containers cannot be nested.

<!--
type: tab
title: My First Tab
-->

The contents of tab 1.

<!--
type: tab
title: My Second Tab
-->

The contents of tab 2.

<!-- type: tab-end -->

```md title="Try it out!" 
<!--
type: tab
title: My First Tab
-->

The contents of tab 1.

<!--
type: tab
title: My Second Tab
-->

The contents of tab 2.

<!-- type: tab-end -->
```

## HTML

> Most basic HTML is supported. **However, we highly recommend using the markdown equivalent whenever possible.

Reviewer: ```suggestion
Most basic HTML is supported. **However, we highly recommend using the Markdown equivalent whenever possible.
```

**
<table>
  <thead>
    <tr>
      <td>heading 1</td>
      <td>heading 2</td>
    </tr>
  </thead>

  <tbody>
    <tr>
      <td>hello</td>
      <td>world</td>
    </tr>
  </tbody>
</table>

---

## Embeds

<!-- theme: warning -->

> This hasn't been implemented yet. **[Vote for this feature](https://roadmap.stoplight.io/c/28-rich-embeds-such-as-code-samples-videos-etc)!**

> This is a youtube video:

Reviewer: ```suggestion
This is a YouTube video:
```

Reviewer: I am not sure if there's a reason it's resolved. Maybe by mistake!
